frontend/src/components/profile/Friend.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import styles from './Friend.module.css'
+import IconAddFriend from '../../assets/icon/add_friend.svg'
+import IconRemoveFriend from '../../assets/icon/remove_friend.svg'
+import IconGenericPicture from '../../assets/icon/generic_picture.svg'
+import ClickableIcon from './ClickableIcon'
+
+export interface FriendProps {
+    id: number
+    name: string
+    picture: string
+    status: 'online' | 'offline' | 'playing'
+    isFriend: boolean
+}
+
+const Friend = ({ name, picture, status, isFriend }: FriendProps) => {
+    const getBorderColor = () => {
+        if (!isFriend) return 'var(--color-black-grey)'
+        switch (status) {
+            case 'online':
+                return 'var(--color-purple)'
+            case 'offline':
+                return 'var(--color-black-grey)'
+            case 'playing':
+                return 'var(--color-mid-green)'
+            default:
+                return 'var(--color-white)'
+        }
+    }
+
+    const getOpacity = () => {
+        switch (status) {
+            case 'online':
+                return '100%'
+            case 'playing':
+                return '100%'
+            default:
+                return '70%'
```

Comment:
It is probably more explicit
> see: [should-switch-statements-always-contain-a-default-clause](https://stackoverflow.com/questions/4649423/should-switch-statements-always-contain-a-default-clause)
```suggestion
            case 'online':
                return '100%'
            case 'playing':
                return '100%'
            case 'offline':
                return '70%'
```

Reply:
Absolutely, good review!